crates/nassun/src/fetch/npm.rs

```diff
@@ -145,3 +145,150 @@ impl PackageFetcher for NpmFetcher {
+            name: "npm".to_string(),
+            requested: Some(VersionSpec::Range(Range::parse(">1.0.0").unwrap())),
+        };
+        let cache_path = tempdir()?;
+        let name = fetcher.name(&spec, cache_path.path()).await?;
+        assert_eq!(name, "npm");
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn read_packument() -> Result<()> {
+        let mut mock_server = mockito::Server::new();
+        let example_response = format!(
+            r#"{{
+            "_attachments": {{}},
+            "_id": "oro-test-example",
+            "_rev": "1-0da57asf9e977d423ed4a35aedc17d250",
+            "author": {{
+                "email": "oro-test@example.com",
+                "name": "Orogene Test"
+            }},
+            "description": "Example Orogene package",
+            "dist-tags": {{
+                "latest": "1.0.0"
+            }},
+            "license": "ISC",
+            "maintainers": [
+                {{
+                    "email": "oro-test@example.com",
+                    "name": "oro-test"
+                }}
+            ],
+            "name": "oro-test-example",
+            "readme": "Nothing much",
+            "readmeFilename": "README.md",
+            "time": {{
+                "1.0.0": "2023-03-13T12:33:00.044Z",
+                "created": "2023-03-13T12:33:00.045Z",
+                "modified": "2023-03-13T12:33:00.046Z"
+            }},
+            "versions": {{
+                "1.0.0": {{
+                    "_from": ".",
+                    "_id": "oro-test-example@1.0.0",
+                    "_nodeVersion": "1.5.0",
+                    "_npmUser": {{
+                        "email": "oro-test@example.com",
+                        "name": "oro-test"
+                    }},
+                    "_npmVersion": "2.7.0",
+                    "_shasum": "bbf102d5ae73afe2c553295e0fb02230216f65b1",
+                    "author": {{
+                        "email": "oro-test@example.com",
+                        "name": "oro-test"
+                    }},
+                    "description": "The first version of the Orogene Test Example",
+                    "directories": {{}},
+                    "dist": {{
+                        "shasum": "bbf102d5ae73afe2c553295e0fb02230216f65b1",
+                        "tarball": "{}/oro-test-example/-/oro-test-example-1.0.0.tgz"
+                    }},
+                    "license": "ISC",
+                    "main": "index.js",
+                    "maintainers": [
+                        {{
+                            "email": "oro-test@example.com",
+                            "name": "oro-test"
+                        }}
+                    ],
+                    "name": "oro-test-example",
+                    "scripts": {{
+                        "test": "echo \"Error: no test specified\" && exit 1"
+                    }},
+                    "version": "1.0.0"
+                }}
+            }}
+        }}"#,
```

Comment:
it looks like you published an actual package just to see what got published, and tbh I think that's a great idea. I appreciate using a "real" packument to test this, rather than a subset of the data....

Reply:
I grabbed an example from the `tiny-tarball` package referenced in https://github.com/npm/registry/blob/master/docs/responses/package-metadata.md
Just changed the names a bit to make it clear that this is not supposed to reference a real registry entry.